greatarc/location.py

```diff
@@ -9,9 +9,16 @@ def __init__(self, name, latitude, longitude):
     def get_coordinates(self):
         return f"Latitude: {self.latitude}, Longitude: {self.longitude}"
 
-    def calculate_distance(self, other_location):
+    def calculate_distance(self, other_location, unit='km'):
+        
         import math
         # Haversine formula to calculate distance between two points on the Earth
+
+        if unit == "km":
+            radius = 6371
+
+        elif unit == "miles":
+            radius = 6371 * 0.621371
         
         radius = 6371  # Earth radius in kilometers
```

Comment:
You would need to remove this line, otherwise, radius is 6371 regardless of unit specified.